src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -23380,15 +23436,62 @@ public C()
                 //     public C()
                 Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "field3").WithLocation(10, 12),
-                // (15,13): warning CS8602: Dereference of a possibly null reference.
-                //             field2.ToString();
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field2").WithLocation(15, 13),
+                // (17,13): warning CS8602: Dereference of a possibly null reference.
+                //             field4.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field4").WithLocation(17, 13)
+                );
+        }
+
+        [Fact]
+        public void MemberNotNullWhenTrue_Multiple_Property()
+        {
+            var c = CreateNullableCompilation(new[] { @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public string field1;
+    public string? field2;
+    public string? field3;
+    public string? field4;
+
+    public C()
+    {
+        if (Init)
+        {
+            field1.ToString();
+            field2.ToString();
+            field3.ToString();
+            field4.ToString();
+        }
+        else
+        {
+            throw null!;
+        }
+    }
+
+    [MemberNotNullWhen(true, nameof(field1))]
+    [MemberNotNullWhen(true, nameof(field2), nameof(field3))]
+    bool Init
+    {
+        get => true;
+        set { }
+    }
+}
+", MemberNotNullWhenAttributeDefinition }, parseOptions: TestOptions.RegularPreview);
+
+            c.VerifyDiagnostics(
                 // (17,13): warning CS8602: Dereference of a possibly null reference.
                 //             field4.ToString();
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field4").WithLocation(17, 13),
-                // (25,47): error CS0579: Duplicate 'MemberNotNullWhen' attribute
-                //     [MemberNotNullWhen(true, nameof(field1)), MemberNotNullWhen(true, nameof(field2))]
-                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "MemberNotNullWhen").WithArguments("MemberNotNullWhen").WithLocation(25, 47)
+                // (29,16): warning CS8775: Member 'field1' may not have a null value when exiting with `True`.
+                //         get => true;
+                Diagnostic(ErrorCode.WRN_MemberNotNullWhen, "true").WithArguments("field1", "True").WithLocation(29, 16),
+                // (29,16): warning CS8775: Member 'field2' may not have a null value when exiting with `True`.
+                //         get => true;
+                Diagnostic(ErrorCode.WRN_MemberNotNullWhen, "true").WithArguments("field2", "True").WithLocation(29, 16),
+                // (29,16): warning CS8775: Member 'field3' may not have a null value when exiting with `True`.
+                //         get => true;
+                Diagnostic(ErrorCode.WRN_MemberNotNullWhen, "true").WithArguments("field3", "True").WithLocation(29, 16)
                 );
         }
```

Comment:
>} [](start = 8, length = 1)

Please add a test (if there isn't one already) with multiple applications of the params version of `MemberNotNull`.

Please add a test with `[MemberNotNull(member: null)]` and `[MemberNotNull(members: null)]` including repetitions.

Similarly for `MemberNotNullWhen`.